pkg/monitor/pod_test.go

```diff
@@ -0,0 +1,221 @@
+package monitor
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+func newPod(namespace, name string) *corev1.Pod {
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+	}
+}
+
+func addIP(in *corev1.Pod, ip string) *corev1.Pod {
+	out := in.DeepCopy()
+	out.Status.PodIPs = append(out.Status.PodIPs, corev1.PodIP{IP: ip})
+	return out
+}
+
+func setDeletionTimestamp(in *corev1.Pod, deletionTime time.Time) *corev1.Pod {
+	out := in.DeepCopy()
+	out.DeletionTimestamp = &metav1.Time{Time: deletionTime}
+	return out
+}
+
+func Test_podNetworkIPCache_updatePod(t *testing.T) {
+
+	type fields struct {
+		podIPsToCurrentPodLocators map[string]sets.String
+	}
+	type args struct {
+		pod *corev1.Pod
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   []monitorapi.Condition
+	}{
+		{
+			name: "newly-created",
+			fields: fields{
+				podIPsToCurrentPodLocators: map[string]sets.String{},
+			},
+			args: args{
+				pod: addIP(newPod("alfa", "zulu"), "10.28.0.96"),
```

Comment:
Suspect your code is fine but just incase anyone does anything funky in the future, it's probably worth sprinkling some ipv6 addresses in some tests.